### ISSUE TITLE
Error sending a message when coming back from background and locked state

### ISSUE THREAD

**Issue by MilanNosal** [twilio/ip-demo-ios#46]
When I send the app into background, lock the screen, and let the device inactive for some longer period of time (at least couple of minutes), when I come back, I am not able to send a message. The call comes back unsuccessful, result text is empty, the result code is 101, and this is what I get when I try to print out `result.error.debugDescription`:

    Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=}

I am not able to find anything on the web for this error code, and since there is nothing else (no result text, nor error description), I am getting desperate.

**Comment by berkus** [comment]
hi, could you please post entire SDK log?

**Comment by vidhyasri2106** [comment]
Same issue for me. Is it possible to send a Pdf or Text file using twilio chat api in iOS?

**Comment by berkus** [comment]
@vidhyasri2106 sorry, how is file sending related to this issue?

**Comment by vidhyasri2106** [comment]
Sorry for bad explanation. In my case, i am getting the same error but when i try to send a file like pdf using media stream

**Comment by berkus** [comment]
Could you also please post the log when this happens?

**Comment by vidhyasri2106** [comment]
Please find the log captured below, while i try to send a file


```
2019-03-07 16:35:54.678101+0530 ChatQuickstart[4251:795873] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2019-03-07 16:35:54.679118+0530 ChatQuickstart[4251:795873] [MC] Reading from public effective user settings.
2019-03-07 16:35:55.065130+0530 ChatQuickstart[4251:795991] <<<<TwilioChatClient: 0x149a00f90>>>> TNRegNotificationCtxSession[2]:        0x16d1e7000 | 03/07/16:35:55.065 | WARNING  | 17     | TNRegNotificationCtxSession | Could not operate with notification context in twilsock. Error 9
2019-03-07 16:35:55.065444+0530 ChatQuickstart[4251:796008] <<<<TwilioChatClient: 0x149a00f90>>>> TNWebsocket[1]:        0x16db33000 | 03/07/16:35:55.065 | CRITICAL | 7      | TNWebsocket | Received error while waiting for message: Net Exception
2019-03-07 16:35:55.065793+0530 ChatQuickstart[4251:795991] <<<<TwilioChatClient: 0x149a00f90>>>> TNFinalState[2]:        0x16d1e7000 | 03/07/16:35:55.065 | WARNING  | 53     | TNFinalState | Unexpected event: Error
2019-03-07 16:35:55.975816+0530 ChatQuickstart[4251:796351] [DocumentManager] Failed to associate thumbnails for picked URL file:///private/var/mobile/Containers/Data/Application/478E4BB4-9F7D-4856-9503-B85D0AC9E860/Documents/ps_kofax-mobile-sdk_en.pdf with the Inbox copy file:///private/var/mobile/Containers/Data/Application/F33C66FF-0DC2-44BB-BE61-7DD08375D2E4/tmp/com.twilio.ChatQuickstartNew-Inbox/ps_kofax-mobile-sdk_en.pdf: Error Domain=QLThumbnail Code=2 "(null)" UserInfo={NSUnderlyingError=0x282ea0450 {Error Domain=GSLibraryErrorDomain Code=3 "Generation not found" UserInfo={NSDescription=Generation not found}}}
Creation failed: Optional(Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=})
```

And the code i am using below,

 ```
do {
                let fileData = try Data(contentsOf: pickedDocUrl)
                // Prepare the upload stream and parameters
                let messageOptions = TCHMessageOptions()
                let inputStream = InputStream(data: fileData)
                messageOptions.withMediaStream(inputStream,
                                               contentType: "application/pdf",
                                               defaultFilename: "\(pickedDocUrl.lastPathComponent.components(separatedBy: ".")[0]).pdf",
                    onStarted: {
                        // Called when upload of media begins.
                        print("Media upload started")
                },
                    onProgress: { (bytes) in
                        // Called as upload progresses, with the current byte count.
                        print("Media upload progress: \(bytes)")
                }) { (mediaSid) in
                    // Called when upload is completed, with the new mediaSid if successful.
                    // Full failure details will be provided through sendMessage's completion.
                    print("Media upload completed")
                }

                // Trigger the sending of the message.
                self.generalChannel?.messages?.sendMessage(with: messageOptions,
                                                           completion: { (result, message) in
                                                            self.pickedDocUrl = nil

                                                            if !result.isSuccessful() {
                                                                print("Creation failed: \(String(describing: result.error))")
                                                            } else {
                                                                print("Creation successful")
                                                            }
                })
            } catch {
                print("Unable to load data: \(error)")
            }
```

**Comment by berkus** [comment]
Can you please post the entire log with verbose level? (Run [setLogLevel:](https://media.twiliocdn.com/sdk/ios/chat/releases/2.4.5/docs/Classes/TwilioChatClient.html#//api/name/setLogLevel:) with log level [TCHLogLevelTrace](https://media.twiliocdn.com/sdk/ios/chat/releases/2.4.5/docs/Constants/TCHLogLevel.html) before constructing TwilioChatClient).

**Comment by vidhyasri2106** [comment]
How and where to set this log level? There is no trace option found.  Only Fatal, Critical, Info, Warning, Debug are present.

**Comment by vidhyasri2106** [comment]
@berkus Please find the complete log set to debug below,

[text.txt](https://github.com/twilio/twilio-chat-demo-ios/files/2981902/text.txt)

**Comment by berkus** [comment]
Uh, this looks like an OLD sdk.

**Comment by vidhyasri2106** [comment]
@berkus So will it work if i try with new SDK? I want to know if it is possible to send pdf file or text file using Twilio Chat SDK. In the documentation, example is given only for image not pdf or text.

**Comment by berkus** [comment]
It should be possible, as far as I know the Media can be anything at all. Our tests send millions of `AAAAAAA` characters and it works fine.

**Comment by vidhyasri2106** [comment]
Okay. I will try updating the library and see. Thanks!

**Comment by vidhyasri2106** [comment]
@berkus I have updated to new sdk(TwilioChatClient 2.4.5) but still seeing the same error on trying to upload a pdf :( Below is the log file.

[text (1).txt](https://github.com/twilio/twilio-chat-demo-ios/files/2983317/text.1.txt)

**Comment by berkus** [comment]
I do not see any attempts to send or receive media in this log file.

Can you write down step by step what you do in this scenario?

Do you get "Media upload started" message from your code?

If not, what is the error printed from this code?:
```
catch {
               print("Unable to load data: \(error)")
           }
```

**Comment by vidhyasri2106** [comment]
Below is the code i am using to upload a PDF. On sendMessage completion,  since the result is not success, it throws **"Creation failed: Optional(Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=})
"**
Also it doesn't go inside Media Upload loop at all.

```
do {
               let fileData = try Data(contentsOf: pickedDocUrl)
               // Prepare the upload stream and parameters
               let messageOptions = TCHMessageOptions()
               let inputStream = InputStream(data: fileData)
               messageOptions.withMediaStream(inputStream,
                                              contentType: "application/pdf",
                                              defaultFilename: "\(pickedDocUrl.lastPathComponent.components(separatedBy: ".")[0]).pdf",
                   onStarted: {
                       // Called when upload of media begins.
                       print("Media upload started")
               },
                   onProgress: { (bytes) in
                       // Called as upload progresses, with the current byte count.
                       print("Media upload progress: \(bytes)")
               }) { (mediaSid) in
                   // Called when upload is completed, with the new mediaSid if successful.
                   // Full failure details will be provided through sendMessage's completion.
                   print("Media upload completed")
               }

               // Trigger the sending of the message.
               self.generalChannel?.messages?.sendMessage(with: messageOptions,
                                                          completion: { (result, message) in
                                                           self.pickedDocUrl = nil

                                                           if !result.isSuccessful() {
                                                               print("Creation failed: \(String(describing: result.error))")
                                                           } else {
                                                               print("Creation successful")
                                                           }
               })
           } catch {
               print("Unable to load data: \(error)")
           }
```

**Comment by berkus** [comment]
Code 101 means you are shutting down the client while it's still doing something.

Where is TwilioChatClient created and who owns it?

**Comment by vidhyasri2106** [comment]
This only happens for PDF or Text file. If i try to send a normal text message or image upload, it works just fine. I am creating TwilioChatClient in **ViewDidAppear** method. Media upload is not getting started at all.

**Comment by berkus** [comment]
I used the demo app from this repository, slightly modified it to upload arbitrary files and successfully uploaded a text file just now. No problems at all.

[Here is the patch I applied](https://gist.github.com/berkus/ee28f0c8e114b74ad01f84310b8dfd20) for reference - try using it?

**Comment by vidhyasri2106** [comment]
I see you have used the same mime type("image/jpeg") for uploading other files like Pdf or text as well. I was using "application/pdf" to do file upload. I will test this and let you know if this works!

**Comment by berkus** [comment]
It worked just the same when I changed that to "application/pdf".

**Comment by vidhyasri2106** [comment]
I tried using the same code with swift and having "application/pdf" but it is still not working. Getting the same error. What could be the possible issue here? Can you please give me a sample that has complete code to test this? Also can you please upload the code snippet in swift?

**Comment by vidhyasri2106** [comment]
@berkus I downloaded the complete ObjC project and tested with the patch you provided. I am getting the below error while trying to send a pdf,

`Error Domain=signal.sdk.domain.error Code=400 "Bad Request" UserInfo={kTCHErrorMsgKey=Bad Request, NSLocalizedDescription=Bad Request}`

Below is the ChannelViewController file

[ChannelViewController.rtf.zip](https://github.com/twilio/twilio-chat-demo-ios/files/2986703/ChannelViewController.rtf.zip)

Also can you please provide me with the list of mime types supported for iOS?

**Comment by berkus** [comment]
iOS supports all the mime-types there are, for `UIDocumentPickerController` the supported types are [linked from the documentation](https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/UTIRef/Introduction/Introduction.html#//apple_ref/doc/uid/TP40009257).

Have you created the token for the demo app as described in the readme, and successfully logged in? I do not understand at which point you get 400 response and there are no logs to confirm that.

**Comment by vidhyasri2106** [comment]
Yes i created the token and able to login successfully and send images, text message etc. but not pdf or a text file. I get 400 response in the downloaded ObjC project when i try to send pdf file.
Please find the logs below,

```
2019-03-20 12:33:28.882437+0530 Demo-ObjC[1539:164399] Failed to get token, error: Error Domain=NSCocoaErrorDomain Code=3000 "no valid “aps-environment” entitlement string found for application" UserInfo={NSLocalizedDescription=no valid “aps-environment” entitlement string found for application}
2019-03-20 12:34:21.279283+0530 Demo-ObjC[1539:164399] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2019-03-20 12:34:21.279597+0530 Demo-ObjC[1539:164399] [MC] Reading from public effective user settings.
2019-03-20 12:34:26.286289+0530 Demo-ObjC[1539:164399] Picked file:///private/var/mobile/Containers/Data/Application/478E4BB4-9F7D-4856-9503-B85D0AC9E860/Documents/ps_kofax-mobile-sdk_en.pdf
2019-03-20 12:34:29.723380+0530 Demo-ObjC[1539:164399] error Error Domain=signal.sdk.domain.error Code=400 "Bad Request" UserInfo={kTCHErrorMsgKey=Bad Request, NSLocalizedDescription=Bad Request}

```
Also, can you please confirm if the code i attached is correct?

**Comment by berkus** [comment]
@vidhyasri2106 re:
>  and having "application/pdf" but it is still not working. Getting the same error. What could be the possible issue here?

please check that you don't create TwilioChatClient as a LOCAL var in `ViewDidLoad` because it would mean it is deleted the moment that method ends. I would suspect this is why it's not sending any media, but without seeing your entire source code I cannot tell for sure.

**Comment by vidhyasri2106** [comment]
I am not creating TwilioChatClient as a Local var. Also i am able to successfully send images every-time. I get this issue only while i try to send a file like pdf or text.

**Comment by berkus** [comment]
> I get this issue only while i try to send a file like pdf or text.

This is what makes it hard to believe, because it works the same no matter what data you send. Could you send me your Swift source code privately to berkus@twilio email? I will try to build it locally and verify.

**Comment by vidhyasri2106** [comment]
@berkus  I have sent the sample code to your mail id, Please check

**Comment by berkus** [comment]
I checked your source code.

With a minor change of UIDocumentPicker mode from `import` to `open` everything works fine:

```
Media upload started
Media upload progress: 4096
Media upload progress: 135168
Media upload progress: 139264
Media upload progress: 147456
Media upload progress: 159744
Media upload progress: 180224
Media upload progress: 225280
Media upload progress: 356352
Media upload progress: 589824
Media upload progress: 838275
Media upload completed
Channel created.
mediaFilename: Optional("gpce17-gpce17main54_Reducing Calling Convention Overhead in Object-Oriented Programming on Embedded ARM Thumb-2 Platforms.pdf") (optional)
mediaSize: 838275
```

**Comment by vidhyasri2106** [comment]
@berkus Can you please upload the updated code or file? I tried changing it to open but it is not working for me :( Just want to check if i am doing something wrong.

Getting below error again,

`Creation failed: Optional(Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=})`

**Comment by berkus** [comment]
I sent you a PR.

**Comment by vidhyasri2106** [comment]
Thank you!! I tried running the updated source code and got the below error,

`Unable to load data: Error Domain=NSCocoaErrorDomain Code=257 "The file “sample.pdf” couldn’t be opened because you don’t have permission to view it." UserInfo={NSFilePath=/private/var/mobile/Containers/Data/Application/478E4BB4-9F7D-4856-9503-B85D0AC9E860/Documents/sample.pdf, NSUnderlyingError=0x282d646c0 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}`

If i try to fix this by adding `pickedDocUrl.startAccessingSecurityScopedResource()` before creating the media stream i am getting the below error,

`Creation failed: Optional(Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=})
`

Is it some issue with the iPhone? Can you please let me know from where you selecting the file? Is it from iCloud or On My iPhone?

**Comment by berkus** [comment]
Are you running this on an actual device? Could you try testing it on the simulator? We then could see if this is iphone problem or something else.

**Comment by vidhyasri2106** [comment]
It works sometimes in Simulator but most of the times getting the same error as mentioned above.

**Comment by berkus** [comment]
Let me investigate that on my devices.

**Comment by vidhyasri2106** [comment]
One more observation. Whenever i try to select a pdf in simulator or device using DocumentPicker, i am getting the below error and then after some time the channels are getting created and established again as shown below,

> Creation failed: Optional(Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=})
channelsList Optional(<TCHMessages: 0x600002c1b270>)
messages Optional([])
mediaFilename: Optional("sample.pdf") (optional)
mediaSize: 3028
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/sample.pdf
2019-03-21 15:07:16.389454+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("FMKYA.png") (optional)
mediaSize: 256337
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/FMKYA.png
2019-03-21 15:07:16.391204+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("ZEGLR.png") (optional)
mediaSize: 339190
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/ZEGLR.png
FILE AVAILABLE
Channel joined with result <TCHResult: 0x600002265fa0>
mediaFilename: Optional("sample.pdf") (optional)
mediaSize: 3028
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/sample.pdf
2019-03-21 15:07:16.398870+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("FMKYA.png") (optional)
mediaSize: 256337
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/FMKYA.png
2019-03-21 15:07:16.400766+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("ZEGLR.png") (optional)
mediaSize: 339190
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/ZEGLR.png
FILE AVAILABLE
Creation failed: Optional(Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=})
channelsList Optional(<TCHMessages: 0x600002c97600>)
messages Optional([])
mediaFilename: Optional("sample.pdf") (optional)
mediaSize: 3028
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/sample.pdf
2019-03-21 15:07:53.837902+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("FMKYA.png") (optional)
mediaSize: 256337
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/FMKYA.png
2019-03-21 15:07:53.839629+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("ZEGLR.png") (optional)
mediaSize: 339190
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/ZEGLR.png
FILE AVAILABLE
Channel joined with result <TCHResult: 0x600002275a80>
mediaFilename: Optional("sample.pdf") (optional)
mediaSize: 3028
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/sample.pdf
2019-03-21 15:07:54.113187+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("FMKYA.png") (optional)
mediaSize: 256337
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/FMKYA.png
2019-03-21 15:07:54.114976+0530 ChatQuickstart[31588:3029205] [framework] BOMStream BOMStreamWithFileAndSys(int, off_t, size_t, int, char *, BomSys *): read: No such file or directory
mediaFilename: Optional("ZEGLR.png") (optional)
mediaSize: 339190
tempFilename /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Data/Application/D7F7F4B5-B3E8-4658-938B-42275F1DF703/tmp/ZEGLR.png
FILE AVAILABLE

**Comment by vidhyasri2106** [comment]
@berkus  I think the issue is not with uploading a pdf or text file. Even if i comment the whole code out inside **didPickDocumentAt** delegate method and then try to just select a document from document picker in simulator/device, i get the same error. Somehow the twilio chat session is getting error every-time we select a document from document picker, no matter whether we try to upload the selected document or not.

**Comment by berkus** [comment]
Chat SDK has no idea about document picker or any UI stuff at all.

This must be something else.

What happens if you leave the document picker open for a while and not choose any document? Do you get the same error eventually?

**Comment by vidhyasri2106** [comment]
I am not getting any error while i am in document picker screen no matter how much time. Only if i dismiss  it by selecting a document or by clicking **cancel**, i get the error.

**Comment by vidhyasri2106** [comment]
I commented all the code inside **didPickDocumentAt** delegate method and added the below code to simply send a text message. I am still getting the same error. 

       if let messages = self.generalChannel?.messages {
            let messageOptions = TCHMessageOptions().withBody("test")
            print("messageOptions \(messageOptions)")
            messages.sendMessage(with: messageOptions, completion: { (result, message) in
                print("result \(String(describing: result.error))")
                print("message \(String(describing: message))")
            })
        }

Console below,

```
channelsList Optional(<TCHMessages: 0x6000029a63a0>)
messages Optional([])
Channel joined with result <TCHResult: 0x6000027aa8e0>
2019-03-21 15:39:18.799259+0530 ChatQuickstart[33476:3089009] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /Users/vidhyasri.so/Library/Developer/CoreSimulator/Devices/BBE522C5-200F-4257-B227-26AEA2D3BED1/data/Containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2019-03-21 15:39:18.799605+0530 ChatQuickstart[33476:3089009] [MC] Reading from private effective user settings.
messageOptions <TCHMessageOptions: 0x6000027bb220>
result <TCHResult: 0x6000027baec0>
message Optional(<<TCHMessage: 0x6000027b12a0>: sid: 'IM160bf4e513a54da8800f5ad922076d47', index: '384', author: 'Vidhya', timestamp: '2019-03-21T10:09:33.031Z', body: 'hi'>)
messageOptions <TCHMessageOptions: 0x600002787580>
result Optional(Error Domain=signal.sdk.domain.error Code=101 "" UserInfo={kTCHErrorMsgKey=, NSLocalizedDescription=})
message nil
channelsList Optional(<TCHMessages: 0x60000297c510>)
messages Optional([<<TCHMessage: 0x600002764d80>: sid: 'IM160bf4e513a54da8800f5ad922076d47', index: '384', author: 'Vidhya', timestamp: '2019-03-21T10:09:33.031Z', body: 'hi'>])
Channel joined with result <TCHResult: 0x60000276e480>
```

**Comment by berkus** [comment]
Could you produce the full SDK log in Verbose (Trace) mode, when this error happens?

**Comment by vidhyasri2106** [comment]
Please find the complete log below,

[Console.txt](https://github.com/twilio/twilio-chat-demo-ios/files/2991687/Console.txt)

**Comment by berkus** [comment]
Hey, in that log i clearly see the TwilioChatClient is *shut down* and then recreated. The error returned exactly after it gets shut down. Double check the code, there might be some variable reuse or something - e.g. viewDidAppear is called again after the document has been picked.

**Comment by berkus** [comment]
Use chat client management approach from this demo repo (a separate sharedInstance of ChatManager that exists regardless of the view you are currently in).

**Comment by vidhyasri2106** [comment]
My bad, Turns out **Logout** was called in the sample code i downloaded in **viewDidDisappear** that causes the **TwilioChatClient** to shut down. It started working after i commented it out. 
Thank you so much for your support!!

**Comment by berkus** [comment]
You also have `login()` in viewDidAppear - this means your client will be recreated after you open the view each time. Better would be to put it in `viewDidLoad`, even better - move it to ChatManager.

**Comment by berkus** [comment]
@MilanNosal for your case I think we found the root cause (it's not the same as with @vidhyasri2106 I believe) and are working on releasing a fix ASAP.

**Comment by berkus** [comment]
@MilanNosal I believe your problem should be fixed by release `2.4.6` of the ios chat sdk. Please, test!
